CHANGELOG.md

```diff
@@ -14,6 +15,7 @@
 * [ENHANCEMENT] ui: Make "hide empty rules" and hide empty rules" persistent #15807
 * [ENHANCEMENT] web/api: Add a limit parameter to `/query` and `/query_range`. #15552
 * [ENHANCEMENT] api: Add fields Node and ServerTime to `/status`. #15784
+* [PERF] Scraping: defer computing labels for dropped targets until they are needed by the UI.  #15261
```

Comment:
```suggestion
* [ENHANCEMENT] Scraping: defer computing labels for dropped targets until they are needed by the UI.  #15261
```

Reply:
Comes from https://github.com/prometheus/prometheus/pull/15901#issuecomment-2639550632

@bboreham wdyt?

Reply:
I'm inclined to keep this under `[PERF]`. https://github.com/prometheus/prometheus/pull/15261 is labeled optimization as well.